app/models/cookbook.rb

```diff
@@ -351,21 +350,23 @@ def download_count
   #
   # A cookbook can only be replaced with a cookbook that is not deprecated.
   #
-  # @param replacement_cookbook [Cookbook] the cookbook to succeed this cookbook
-  #   once deprecated
+  # @param replacement_cookbook_name [String] the name of the cookbook to
+  #   succeed this cookbook once deprecated
   #
   # @return [Boolean] whether or not the cookbook was successfully deprecated
   #   and  saved
   #
-  def deprecate(replacement_cookbook)
-    if replacement_cookbook.deprecated?
+  def deprecate(replacement_cookbook_name = '')
+    replacement_cookbook = Cookbook.find_by_name replacement_cookbook_name
```

Comment:
Opted for `find_by_name` even though it does not do `with_name`'s lowercase searching. `find_by_name` returns the first match or a nil for us. Odds are that the incoming `replacement_cookbook_name` will match the existing cookbook because the web form was populated by a search of the database.